### DISCUSSION TITLE
Array Specifics

### DISCUSSION THREAD

**Discussion by GoldenStack** [TodePond/GulfOfMexico#422] [Q&A] [answered]
Do float indices make arrays act similar to maps (where you can provide any key) or do they add the element in the middle and shift all the others over?
so basically:
```java
const var scores = [3, 2, 5]!
scores[0.5] = 4!
print(scores)! //[3, 2, 4, 5]
```
Are the elements of this array indexed as `[-1, 0, 0.5, 1]`, or `[-1, 0, 1, 2]`?

**Comment by mybearworld** [comment, marked as answer]
> You can now use floats for indexes too!

Floats are used as indices. So, this would be `[-1, 0, 0.5, 1]`.
